source/solvers/tracer_assemblers.cc

```diff
@@ -698,3 +698,98 @@ TracerAssemblerBoundaryNitsche<dim>::assemble_rhs(
 
 template class TracerAssemblerBoundaryNitsche<2>;
 template class TracerAssemblerBoundaryNitsche<3>;
+
+template <int dim>
+void
+TracerAssemblerReaction<dim>::assemble_matrix(
+  const TracerScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData    &copy_data)
+{
+  // Scheme and physical properties
+  const std::vector<double> &k = scratch_data.tracer_reaction_constant;
+  const double               a = scratch_data.tracer_reaction_order;
+
+  // Loop and quadrature informations
+  const auto        &JxW_vec    = scratch_data.JxW;
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  auto &strong_jacobian_vec = copy_data.strong_jacobian;
+  auto &local_matrix        = copy_data.local_matrix;
+
+  // assembling local matrix
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    {
+      const double reaction_coeff =
+        k[q] * a *
+        std::pow(scratch_data.tracer_values[q], a - 1); // k * a * C^(a-1)
```

Comment:
Why do you hardcode the jacobian in the assembler? The physical property itself can provide the jacobian you don,t need to hardcode it. You just need to get the joacbian from the reaction constant no?

Or was your idea just to use a constant that could eventually depend on (T) and etc., and the order itself is a seperate parameter? It's not a bad idea, but document it correctly. I would have assumed that you would have lumped the order effect in the constant (to me it would have been clearer and more readable to be honest). However, if you really think it's clearer this way, it'S fine, just document it very well.